src/interactions/muller_brown.jl

```diff
@@ -0,0 +1,85 @@
+export MullerBrown
+
+@doc raw"""
+    MullerBrown(A, a, b, c, x₀, y₀)
+
+The Muller-Brown potential energy surface, witth 3 minima and 2 saddle points is given by:
+```math
+V(x,y) = \sum_{n=1}^{4} A_kexp[a_k(x-x_k^0)^2 + b_k(x-x_k^0)(y-y_k^0) + c_k(y-y_k^0)^2]
+```
+where A, a, b, c, $x_0$, $y_0$ are 4 element arrays
+"""
+#To-Do
+# A should have energy_units, in general are units working
+# Test coverage
+# Example for documentation
+# Should potential return just a number?
+
+# General interaction:
+# https://juliamolsim.github.io/Molly.jl/dev/docs/#General-interactions
+struct MullerBrown{D,T,L,F,E}
+    A::D # Array w/ units of energy
+    a::T # Array w/ units of 1/L^2
+    b::T # Array w/ units of 1/L^2
+    c::T # Array w/ units of 1/L^2
+    x₀::L # Array w/ units of L
+    y₀::L # Array w/ units of L
+    force_units::F
+    energy_units::E
+end
+
+
+function MullerBrown(A, a, b, c, x₀, y₀; 
+                        force_units = u"kJ * mol^-1 * nm^-1",
+                        energy_units = u"kJ * mol^-1")
+
+    if length(A) != 4 || length(a) != 4 || length(b) != 4 || length(c) != 4 || length(x₀) != 4 || length(y₀) != 4
+        throw(ArgumentError("The length of each array passed to the Muller-Brown potential should be 4."))
+    end
+
+    return MullerBrown{typeof(A), typeof(a), typeof(x₀), typeof(force_units), typeof(energy_units)}(
+        A, a, b, c, x₀, y₀, force_units, energy_units)
+end
+
+#Total potential energy of system
+@inline @inbounds function potential_energy(inter::MullerBrown{D,T,L}, sys, neighbors=nothing) where {D,T,L}
+    return sum(potential.(Ref(inter),sys.coords))
+end
+
+@inline @inbounds @fastmath function potential(inter::MullerBrown{D,T,L}, coord::SVector{2}) where {D,T,L}
```

Comment:
Since `potential` is specific to Muller-Brown it might be better named as `potential_muller_brown`. Also applies to `force`. You can access it in tests with `Molly.potential_muller_brown` - this is required since it isn't exported.